File: Python/G5/G5.py
from pathlib import Path

def a_inicio_b_final(s: str) -> bool:
    if not all(c in "ab" for c in s):
        return False
    return len(s) >= 2 and s[0] == 'a' and s[-1] == 'b'

def main():
    archivo = Path("G5.txt")
    if not archivo.exists():
        print("[ADVERTENCIA] No se encontró G5.txt")
        return
    with archivo.open("r", encoding="utf-8") as f:
        for i, linea in enumerate(f, start=1):
            s = linea.strip()
            acepta = a_inicio_b_final(s)
            veredicto = "acepta" if acepta else "NO acepta"
            mostrar = s if s else "ε"
            print(f"[G5][{i}] {mostrar} => {veredicto}")


main()

File: C/G5/G5.c
#include <stdio.h>
#include <string.h>
#include <stdbool.h>

bool a_inicio_b_final(const char *s) {
    int len = strlen(s);
    if (len < 2) return false;
    for (int i = 0; i < len; i++) {
        if (s[i] != 'a' && s[i] != 'b') return false;
    }
    return s[0] == 'a' && s[len-1] == 'b';
}

void chomp(char *s) {
    int n = strlen(s);
    while (n > 0 && (s[n-1] == '\n' || s[n-1] == '\r')) {
        s[n-1] = '\0';
        n--;
    }
}

int main() {
    FILE *f = fopen("G5.txt", "r");
    if (!f) { printf("[ADVERTENCIA] No se encontró G5.txt\n"); return 0; }
    char linea[256];
    int i = 0;
    while (fgets(linea, sizeof(linea), f)) {
        chomp(linea);
        i++;
        bool acepta = a_inicio_b_final(linea);
        printf("[G5][%d] %s => %s\n", i, linea[0] ? linea : "ε", acepta ? "acepta" : "NO acepta");
    }
    fclose(f);
    return 0;
}
